docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,302 @@
+function, which retrieves the configuration from the hyperparameters. You can then pass this
+configuration to deepspeed.initialize as usual:
+
+.. code:: python
+
+   from determined.pytorch.deepspeed import DeepSpeedTrial, DeepSpeedTrialContext
+   from determined.pytorch import dsat
+
+
+   class MyDeepSpeedTrial(DeepSpeedTrial):
+     def __init__(self, context: DeepSpeedTrialContext) -> None:
+         self.hparams = self.context.get_hparams()
+         config = dsat.get_ds_config_from_hparams(self.hparams)
+         model = ...
+         model_parameters= ...
+
+         model_engine, optimizer, train_loader, lr_scheduler = deepspeed.initialize(
+             model=model, model_parameters=model_parameters, config=config
+         )
+
+Using Determined's DeepSpeed Autotune with a :class:`~determined.pytorch.deepspeed.DeepSpeedTrial`
+instance requires no further changes to your user code.
+
+A full example which uses DeepSpeed Autotune with ``DeepSpeedTrial`` can be found in the `Determined
+GitHub Repo
+<https://github.com/determined-ai/determined/tree/master/examples/deepspeed_autotune/torchvision/deepspeed_trial>`__
+under ``examples/deepspeed_autotune/torchvision/deepspeed_trial`` .
+
+Core API
+========
+
+When using DeepSpeed Autotune with a Core API experiment, there is one additional change to be made
+following the steps in the :ref:`using_deepspeed_trial` section above.
+
+The ``forward``, ``backward``, and ``step`` methods of the ``DeepSpeedEngine`` class need to be
+wrapped in the :func:`~determined.pytorch.dsat.dsat_reporting_context` context manager. This
+addition ensures that the autotuning metrics from each trial are captured and reported back to the
+Determined master.
+
+Here is an example sketch of ``dsat`` code with Core API:
+
+.. code:: python
+
+   for op in core_context.searcher.operations():
+      for (inputs, labels) in trainloader:
+          with dsat.dsat_reporting_context(core_context, op): # <-- The new code
+              outputs = model_engine(inputs)
+              loss = criterion(outputs, labels)
+              model_engine.backward(loss)
+              model_engine.step()
+
+In this code snippet, ``core_context`` is the :class:`~determined.core.Context` instance which was
+initialized with :func:`determined.core.init`. The context manager requires access to both
+``core_context`` and the current :class:`~determined.core.SearcherOperation` instance (``op``) to
+appropriately report results. Outside of a ``dsat`` context, ``dsat_reporting_context`` is a no-op,
+so there is no need to remove the context manager after the ``dsat`` trials have completed.
+
+A full example which uses DeepSpeed Autotune with Core API can be found in the `Determined GitHub
+Repo
+<https://github.com/determined-ai/determined/tree/master/examples/deepspeed_autotune/torchvision/core_api>`__
+under ``examples/deepspeed_autotune/torchvision/core_api``.
+
```

Comment:
For a complete example of how to use DeepSpeed Autotune with Core API, visit the `Determined GitHub
Repo
<https://github.com/determined-ai/determined/tree/master/examples/deepspeed_autotune/torchvision/core_api>`__
and navigate to ``examples/deepspeed_autotune/torchvision/core_api``.